tests/path_tracing/test_path_tracing_midpoint.py

```diff
@@ -0,0 +1,422 @@
+                break
+
+        selected_peer_ip_ifaces_pairs = []
+        rif_rx_ifaces = None
+        if namespace_with_min_two_ip_interface is not None:
+            for v in namespace_neigh_cnt_map[namespace_with_min_two_ip_interface]:
+                selected_peer_ip_ifaces_pairs.append(peer_ip_ifaces_pair_list[v[0]][v[1]])
+                if not rif_rx_ifaces:
+                    if v[0]:
+                        rif_rx_ifaces = \
+                            list(pc_ports_map.keys())[list(pc_ports_map.values())
+                                                      .index(selected_peer_ip_ifaces_pairs[0][1])]
+                    else:
+                        rif_rx_ifaces = selected_peer_ip_ifaces_pairs[0][1][0]
+        else:
+            pytest.skip("Skip test as not enough neighbors/ports.")
+
+        # use first port of first peer_ip_ifaces pair as input port
+        # all ports in second peer_ip_ifaces pair will be output/forward port
+        ptf_port_idx = mg_facts["minigraph_ptf_indices"][selected_peer_ip_ifaces_pairs[0][1][0]]
+        ptf_port_idx_namespace = namespace_with_min_two_ip_interface
+        asic_id = duthost.get_asic_id_from_namespace(ptf_port_idx_namespace)
+        ingress_router_mac = duthost.asic_instance(asic_id).get_router_mac()
+
+        # Some platforms do not support rif counter
+        try:
+            rif_counter_out = TestPathTracingMidpoint.parse_rif_counters(
+                duthost.command("show interfaces counters rif")["stdout_lines"])
+            rif_iface = list(rif_counter_out.keys())[0]
+            rif_support = False if rif_counter_out[rif_iface]['rx_err'] == 'N/A' else True
+        except Exception as e:
+            logger.info("Show rif counters failed with exception: {}".format(repr(e)))
+            rif_support = False
+
+        for prefix in ["2001:db8:2::1/128"]:
+            duthost.shell(duthost.get_vtysh_cmd_for_namespace(
+                "vtysh -c \"configure terminal\" -c \"ip route {} {} tag 1\""
+                .format(prefix, selected_peer_ip_ifaces_pairs[1][0]), ptf_port_idx_namespace))
+        yield selected_peer_ip_ifaces_pairs, rif_rx_ifaces, rif_support, \
+            ptf_port_idx, pc_ports_map, mg_facts["minigraph_ptf_indices"], ingress_router_mac
+
+        for prefix in ["2001:db8:2::1/128"]:
+            duthost.shell(duthost.get_vtysh_cmd_for_namespace(
+                "vtysh -c \"configure terminal\" -c \"no ip route {} {} tag 1\""
+                .format(prefix, selected_peer_ip_ifaces_pairs[1][0]), ptf_port_idx_namespace))
+
+    def teardown_path_tracing(self, duthost):
+        """
+        teardown Path Tracing after test by disabling Path Tracing on all interfaces
+        :param duthost: DUT host object
+        :return:
+        """
+        logger.info("Disable Path Tracing")
+        self.config_path_tracing(duthost, ifname="Ethernet8", enable=False)
+
+    def simple_ipv6_packet(
```

Comment:
We can use simple_tcpv6 or simple_udpv6 from testutils. Please check the other tests.